src/helpers.ts

```diff
@@ -68,11 +68,22 @@ export async function requiredStatusChecksForBranch(
     })
   ).data
 
+  var checksFromBranchProtection = []
+
+  const rules = (
+    await octokit.rest.repos.getBranchRules({
+      ...github.context.repo,
+      branch: branchName,
+    })
+  ).data
```

Comment:
`branch.protected` is still true for this, right? So please move this in an `if (branch.protected)`.